### PR TITLE
core: Implement sceRandomGetRandomNumber

### PR DESCRIPTION
**Pull request by dima-xd**
Needed for Gran Turismo

### CODE REVIEW

File: src/core/libraries/random/random.cpp [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+#include "common/logging/log.h"
+#include "core/libraries/error_codes.h"
+#include "core/libraries/libs.h"
+#include "random.h"
+
+namespace Libraries::Random {
+
+s32 PS4_SYSV_ABI sceRandomGetRandomNumber(void* buf, size_t size) {
```
Comment (raphaelthegreat) [inline src/core/libraries/random/random.cpp]:
```suggestion
s32 PS4_SYSV_ABI sceRandomGetRandomNumber(u8* buf, size_t size) {
```
No need for the cast

**Review by raphaelthegreat — APPROVED** [review]

**Comment by polybiusproxy** [comment]
> Needed for Gran Turismo

Which game?

**Comment by dima-xd** [comment]
> > Needed for Gran Turismo
> 
> Which game?

I think for both. But I tried only 7th.
GT7 crashes now while trying to unlock nullptr mutex in `scePthreadMutexUnlock`

File: src/core/libraries/random/random.cpp (hunk 1 above)
Comment (dima-xd) [reply src/core/libraries/random/random.cpp]: fixed

**Pull request merged**